go/arrow/util/protobuf_reflect.go

```diff
@@ -237,12 +234,10 @@ func (pfr *ProtobufFieldReflection) asList() protobufListReflection {
 }
 
 func (plr protobufListReflection) getDataType() arrow.DataType {
-	for li := range plr.generateListItems() {
-		return arrow.ListOf(li.getDataType())
-	}
 	pfr := ProtobufFieldReflection{
 		descriptor:    plr.descriptor,
 		schemaOptions: plr.schemaOptions,
+		isListItem:    true,
```

Comment:
this is the key change for lists, protoreflect doesn't come with a get underlying type function instead the isList is set to true, so on the 2nd pass to get the data type we override the isList with isListItem